hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -1130,22 +1143,24 @@ public void run(PRESP resp) {
    * List all region locations for the specific table.
    */
   private CompletableFuture<List<HRegionLocation>> getTableHRegionLocations(TableName tableName) {
-    if (TableName.META_TABLE_NAME.equals(tableName)) {
+    if (ROOT_TABLE_NAME.equals(tableName)) {
       CompletableFuture<List<HRegionLocation>> future = new CompletableFuture<>();
-      addListener(connection.registry.getMetaRegionLocations(), (metaRegions, err) -> {
+      addListener(connection.registry.getMetaRegionLocations(), (rootRegions, err) -> {
         if (err != null) {
           future.completeExceptionally(err);
-        } else if (metaRegions == null || metaRegions.isEmpty() ||
-          metaRegions.getDefaultRegionLocation() == null) {
+        } else if (rootRegions == null || rootRegions.isEmpty() ||
+          rootRegions.getDefaultRegionLocation() == null) {
           future.completeExceptionally(new IOException("meta region does not found"));
         } else {
-          future.complete(Collections.singletonList(metaRegions.getDefaultRegionLocation()));
+          future.complete(Collections.singletonList(rootRegions.getDefaultRegionLocation()));
         }
       });
       return future;
     } else {
-      // For non-meta table, we fetch all locations by scanning hbase:meta table
-      return ClientMetaTableAccessor.getTableHRegionLocations(metaTable, tableName);
+      // For non-meta table, we fetch all locations by scanning catalog table
```

Comment:
Is comment right? Seems like we can get users-space tables and regions in meta too?

Reply:
Yes this needs updating.